### ISSUE TITLE
Sphere (CompoundVisual) does not get opaque to markers

### ISSUE THREAD

**Issue by michaelaye** [vispy/vispy#1060]
Using this example code, and rotating the camera around the scene until the markers are behind the sphere, the markers always stay visible, although I tried to make an opaque sphere. For data exploration, this is highly confusing, so I'm looking for a way to make the sphere visual opaque?
Note, I also tried a RGBA sphere color like (0,0,1,1), but to no avail.

``` python
import sys
import numpy as np
from vispy import scene
from vispy.scene import visuals

# create particle data

z = np.linspace(0.8, 2, 100)
x = np.zeros_like(z)
y = np.zeros_like(z)

data = np.vstack([x, y, z]).T
#
# Make a canvas and add simple view
#
canvas = scene.SceneCanvas(keys='interactive', show=True, bgcolor='w')

view = canvas.central_widget.add_view()
# view.camera = 'turntable'
view.camera = 'arcball'
view.camera.fov = 60

# create Enceladus
enc = visuals.Sphere(radius=0.8, color=(0, 0, 1),
                     edge_color=(1, 0, 0), parent=view.scene)

# add a colored 3D axis for orientation
axis = visuals.XYZAxis(parent=view.scene, width=1)

scatter = visuals.Markers()

scatter.set_data(data, edge_color=None, face_color=(0.6, 0.5, 0.4), size=5)

view.add(scatter)


if __name__ == '__main__' and sys.flags.interactive == 0:
    canvas.app.run()
```

**Comment by nilsbecker** [comment]
can confirm the behavior

**Comment by rougier** [comment]
In `visual.Markers.__init__` this line:
```
self.set_gl_state(depth_test=False, blend=True,
                  blend_func=('src_alpha', 'one_minus_src_alpha'))
```

indicate to not use depth testing, that might be the reason. Can you try `depth_test=True` to check if this is the problem ?

**Comment by michaelaye** [comment]
yes, that did it. I suggest this change in the markers visual to provide control over it from the scene level:

```diff
diff --git c/vispy/visuals/markers.py w/vispy/visuals/markers.py
index 6bb37e7..649c9e4 100644
--- c/vispy/visuals/markers.py
+++ w/vispy/visuals/markers.py
@@ -481,7 +481,7 @@ marker_types = tuple(sorted(list(_marker_dict.keys())))
 class MarkersVisual(Visual):
     """ Visual displaying marker symbols.
     """
-    def __init__(self, **kwargs):
+    def __init__(self, depth_test=False, **kwargs):
         self._vbo = VertexBuffer()
         self._v_size_var = Variable('varying float v_size')
         self._symbol = None
@@ -492,7 +492,7 @@ class MarkersVisual(Visual):
         Visual.__init__(self, vcode=vert, fcode=frag)
         self.shared_program.vert['v_size'] = self._v_size_var
         self.shared_program.frag['v_size'] = self._v_size_var
-        self.set_gl_state(depth_test=False, blend=True,
+        self.set_gl_state(depth_test=depth_test, blend=True,
                           blend_func=('src_alpha', 'one_minus_src_alpha'))
         self._draw_mode = 'points'
lines 1-21
```
and then calling the Markers constructor with `depth_test=True` ?
PR?

**Comment by Eric89GXL** [comment]
Seems reasonable to me.

**Comment by rougier** [comment]
Yes. Don't forget to add some documentation in the docstring or some comment.

**Comment by michaelaye** [comment]
ok, will do, thanks!

**Comment by kmuehlbauer** [comment]
`depth_test=True` was implemented with #1191, considering this as fixed.
